app/pages/challenges/hooks/useAddChallengeMutation.tsx

```diff
@@ -0,0 +1,19 @@
+import {useMutation, useQueryClient} from "@tanstack/react-query";
+import {apis} from "@/api/initializeApi";
+
+export const useAddChallengeMutation = () => {
+    const queryClient = useQueryClient();
+
+    const mutation = useMutation({
+        mutationFn: ({userId, challengeRequest}: { userId: number, challengeRequest: any }) =>
+            apis().challenge.addChallenge(userId, challengeRequest),
+        onSuccess: async (_, {userId}) => {
+            await queryClient.refetchQueries({queryKey: ["challenges", userId]});
+        }
```

Comment:
why state is not updated ? easy. you have your query under key "challenges" but you are refreshing "challenges {userId}" so you are updating the hook with unexisting key. just update hook as i told by removing userId? and making it userId always and this will work